daily_milestone.go

```diff
@@ -1,29 +1,65 @@
 package main
 
-import "flag"
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
+var logger *log.Logger
+
+//LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+func getProjectID(BaseURL string, Token string, Projectname string, Namespace string) string {
+	urls := "https://" + BaseURL + "/projects"
 	page := 1
+	strPage := strconv.Itoa(page)
+	s := []string{urls, "?page=", strPage}
```

Comment:
Instead of the manual concatenation remember you can use url encoding from the library:
```
u, err := url.Parse(completeURL)
if err != nil {
    log.Fatal(err)
}
q := u.Query()
q.Add("page", page)
u.RawQuery = q.Encode()
// u can be used to do the request now.
```